### PR TITLE
[multitenancy-manager] add controller

### LINKED ISSUE #9091: [multitenancy-manager] separate controller for Projects
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

We need to control each Project separately.

### Proposed Solution

We need to move [this hook](https://github.com/deckhouse/deckhouse/blob/main/ee/modules/160-multitenancy-manager/hooks/project.go) logic and all dependencies to the separate controller.
And then deploy it inside the module

### Additional Information

_No response_

### LINKED ISSUE #9094: [multitenancy-manager] add config flag for namespaces control
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

To use the policy `everything is a project`

### Proposed Solution

Add a flag [here](https://github.com/deckhouse/deckhouse/blob/main/ee/modules/160-multitenancy-manager/openapi/config-values.yaml), something like:

```
allowOrphanNamespaces:
  type: boolean
  default: true
```

If this flag is disabled - prohibit to create a namespace which is not a part of a Project

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ipaqsa**
## Description
It provides a new separate controller for projects.

## Why do we need it, and what problem does it solve?
Fixes #9091
Fixes #9094 

## What is the expected result?
Projects and templates are managed by a new separate controller.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: multitenancy-manager
type: feature
summary: Add separate controller for projects.
impact: There may be problems with backward compatibility.
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ipaqsa. I've started the workflow for label 'e2e/run/gcp'...

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by miklezzzz** [comment]
```
impact: There may be problems with backward compatibility.
```
what sort of backward compatibility problems? :thinking:

File: ee/modules/160-multitenancy-manager/images/multitenancy-manager/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			return nil, err
+		}
+	}
+
+	managerOpts := manager.Options{
+		LeaderElection:          false,
+		Scheme:                  scheme,
+		GracefulShutdownTimeout: pointer.Duration(10 * time.Second),
+		HealthProbeBindAddress:  ":9090",
+		WebhookServer:           webhook.NewServer(webhook.Options{CertDir: "/certs"}),
```
Comment (miklezzzz) [inline ee/modules/160-multitenancy-manager/images/multitenancy-manager/cmd/main.go:121]: nit: if there is no plan to poll the controller's metrics, it makes sense to disable the metrics handler

**Review by miklezzzz — APPROVED** [review]

File: ee/modules/160-multitenancy-manager/images/multitenancy-manager/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+ARG BASE_DISTROLESS
```
[the remark this answers is not in the document]
Comment (ipaqsa) [reply ee/modules/160-multitenancy-manager/images/multitenancy-manager/Dockerfile:1]: what i should use instead ?

File: ee/modules/160-multitenancy-manager/images/multitenancy-manager/cmd/main.go (hunk 1 above)
Comment (ipaqsa) [reply ee/modules/160-multitenancy-manager/images/multitenancy-manager/cmd/main.go:121]: agree

File: ee/modules/160-multitenancy-manager/images/multitenancy-manager/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (miklezzzz) [reply ee/modules/160-multitenancy-manager/images/multitenancy-manager/Dockerfile:1]: werf, I suppose

[the remark this answers is not in the document]
Comment (miklezzzz) [reply ee/modules/160-multitenancy-manager/images/multitenancy-manager/Dockerfile:1]: but it's up to you, i thought a new shiny thing would require a cool wrapper :smiley:

**Pull request merged**